### PR TITLE
Beta7 fixes

### PR DESCRIPTION
**Pull request by tushargupta51**
fixing 3 more issues.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23issuecomment-128456071%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23issuecomment-129598395%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23issuecomment-132321459%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23discussion_r36750942%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23discussion_r36783382%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23discussion_r37316914%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86%23commitcomment-12621639%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86%23commitcomment-12621702%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23issuecomment-128456071%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40brentschmaltz%20%22%2C%20%22created_at%22%3A%20%222015-08-06T17%3A43%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40brentschmaltz%20added%20the%20logging%20messages%20for%20CreateActorValue%20method%20and%20reverted%20the%20changes%20made%20to%20GetAllKeys%20method.%22%2C%20%22created_at%22%3A%20%222015-08-10T20%3A34%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-18T19%3A11%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Commit%20f226885c6b9b820da44048b95b14817e05f81d86%20src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs%205%20801%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86%23commitcomment-12621702%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Thinking%20about%20making%20this%20public%20and%20replacing%20ResolveIssuerSigningKey%20so%20that%20multiple%20keys%20can%20be%20returned%20from%20delegate%2C%20so%20perhaps%20we%20leave%20the%20signature%20alone%20for%20now.%20%5Cr%5CnFor%20example%20if%20a%20token%20arrives%20without%20a%20%27kid%27%20and%20multiple%20keys%20are%20available%20they%20user%20is%20in%20a%20bad%20spot%20because%20our%20delegate%20only%20returns%20a%20single%20key.%22%2C%20%22created_at%22%3A%20%222015-08-10T18%3A13%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs%3AL801%20%28f226885%29%22%7D%2C%20%22Commit%20f226885c6b9b820da44048b95b14817e05f81d86%20src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs%2041%20978%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86%23commitcomment-12621639%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Instead%20of%20logging%20when%20BootstrapContext%20is%20null.%20Log%20from%20where%20the%20encoded%20jwt%20was%20obtained.%20There%20are%20three%20possibilities.%20BootstrapContext.%20jwt.RawData%2C%20this.WriteToken.%20%22%2C%20%22created_at%22%3A%20%222015-08-10T18%3A11%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs%3AL978%20%28f226885%29%22%7D%2C%20%22Pull%2057e1467ded23dd29851085fb192c04f39bf2505c%20src/System.IdentityModel.Tokens/ErrorMessages.cs%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23discussion_r36750942%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Suggest%20adding%20additional%20comment%20%27%28%20as%20string%20%29%27%22%2C%20%22created_at%22%3A%20%222015-08-11T14%3A29%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%2C%20%7B%22body%22%3A%20%22comment%20updated.%22%2C%20%22created_at%22%3A%20%222015-08-11T18%3A49%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens/ErrorMessages.cs%3AL155-166%22%7D%2C%20%22Pull%202e22365bf16c4cf3f6ac7c2c0360480656480da8%20test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs%2013%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211%23discussion_r37316914%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22add%20newline%22%2C%20%22created_at%22%3A%20%222015-08-18T16%3A00%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs%3AL129-142%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/brentschmaltz%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/brentschmaltz'></a>

- [ ] <a href='#crh-comment-Commit f226885c6b9b820da44048b95b14817e05f81d86 src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs 41 978'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86#commitcomment-12621639'>File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs:L978 (f226885)</a></b>
- <a href='https://github.com/brentschmaltz'></a> Instead of logging when BootstrapContext is null. Log from where the encoded jwt was obtained. There are three possibilities. BootstrapContext. jwt.RawData, this.WriteToken.
- [ ] <a href='#crh-comment-Commit f226885c6b9b820da44048b95b14817e05f81d86 src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs 5 801'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/f226885c6b9b820da44048b95b14817e05f81d86#commitcomment-12621702'>File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs:L801 (f226885)</a></b>
- <a href='https://github.com/brentschmaltz'></a> Thinking about making this public and replacing ResolveIssuerSigningKey so that multiple keys can be returned from delegate, so perhaps we leave the signature alone for now.
For example if a token arrives without a 'kid' and multiple keys are available they user is in a bad spot because our delegate only returns a single key.
- [ ] <a href='#crh-comment-Pull 57e1467ded23dd29851085fb192c04f39bf2505c src/System.IdentityModel.Tokens/ErrorMessages.cs 5'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211#discussion_r36750942'>File: src/System.IdentityModel.Tokens/ErrorMessages.cs:L155-166</a></b>
- <a href='https://github.com/brentschmaltz'></a> Suggest adding additional comment '( as string )'
- <a href='https://github.com/tushargupta51'></a> comment updated.
- [ ] <a href='#crh-comment-Pull 2e22365bf16c4cf3f6ac7c2c0360480656480da8 test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs 13'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211#discussion_r37316914'>File: test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs:L129-142</a></b>
- <a href='https://github.com/brentschmaltz'></a> add newline
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211#issuecomment-128456071'>General Comment</a></b>
- <a href='https://github.com/tushargupta51'></a> @brentschmaltz
- <a href='https://github.com/tushargupta51'></a> @brentschmaltz added the logging messages for CreateActorValue method and reverted the changes made to GetAllKeys method.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/211'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by tushargupta51** [comment]
@brentschmaltz

**Comment by tushargupta51** [comment]
@brentschmaltz added the logging messages for CreateActorValue method and reverted the changes made to GetAllKeys method.

File: src/System.IdentityModel.Tokens/ErrorMessages.cs [hunk 1]
```diff
@@ -155,6 +155,11 @@ public static class ErrorMessages
         public const string IDX10709 = "IDX10709: '{0}' is not well formed: '{1}'. The string needs to be in compact JSON format, which is of the form: '<Base64UrlEncodedHeader>.<Base64UrlEndcodedPayload>.<OPTIONAL, Base64UrlEncodedSignature>'.";
         public const string IDX10710 = "IDX10710: Only a single 'Actor' is supported. Found second claim of type: '{0}', value: '{1}'";
         public const string IDX10711 = "IDX10711: actor.BootstrapContext is not a string AND actor.BootstrapContext is not a JWT";
+        public const string IDX10712 = "IDX10712: actor.BootstrapContext is null. Creating the token using actor.Claims.";
+        public const string IDX10713 = "IDX10713: Creating actor value using actor.BootstrapContext";
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens/ErrorMessages.cs]: Suggest adding additional comment '( as string )'

Comment (tushargupta51) [inline src/System.IdentityModel.Tokens/ErrorMessages.cs]: comment updated.

File: test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs [hunk 2]
```diff
@@ -129,5 +129,13 @@ public void ObjectClaims()
...
+
+        [Fact(DisplayName = "JwtPayloadTests: test claim with null value")]
+        public void TestClaimWithNullValue()
+        {
+            JwtPayload jwtPayload = new JwtPayload();
+            jwtPayload.Add("testClaim", null);
+            List<Claim> claims = jwtPayload.Claims as List<Claim>;   // this should not throw
+        }
     }
 }
```
Comment (brentschmaltz) [inline test/System.IdentityModel.Tokens.Jwt.Tests/JwtPayloadTest.cs]: add newline

**Comment by brentschmaltz** [comment]
:shipit:

**Comment by tushargupta51** [comment]
827af47924248cee0a7c44c51e8fa7722cde2015

**Pull request closed without merging**